vars/terraform.groovy

```diff
@@ -0,0 +1,177 @@
+                  sh 'make -C ./tests'
+                }
+              }
+            }
+          })
+        }
+      }
+    }
+
+    parallelStages['production'] = {
+      stage('Production') {
+        agentTemplate(finalConfig.agentContainerImage, {
+          withCredentials(defaultConfig.productionCredentials) {
+            stage('🦅 Generate Terraform Plan') {
+              // When the job is triggered by the daily cron timer, then the plan succeed only if there is no changes found (e.g. no config drift)
+              // For all other triggers, the plan succeed either there are changes or not
+              String tfCliArsPlan=''
+              final String planFileName = 'terraform-plan-for-humans.txt'
+              if (isBuildCauseTimer) {
+                tfCliArsPlan='-detailed-exitcode'
+              }
+              withEnv([
+                // https://www.terraform.io/docs/cli/config/environment-variables.html#tf_cli_args-and-tf_cli_args_name
+                "TF_CLI_ARGS_plan=${tfCliArsPlan}",
+                "PLAN_FILE_NAME=${planFileName}",
+              ]) {
+                getInfraSharedTools(sharedToolsSubDir)
+
+                try {
+                  sh makeCliCmd + ' plan'
+                }
+                finally {
+                  archiveArtifacts planFileName
+                }
+              }
+            }
+
+            if (isBuildOnChangeRequest) {
+              stage('🗣 Notify User on the PR ') {
+                final String planFileUrl = "${env.BUILD_URL}artifact/terraform-plan-for-humans.txt"
+                publishChecks name: "terraform-plan",
+                  title: "Terraform Plan for this Change Request",
+                  text: "The terraform plan for this change request can be found at <${planFileUrl}>.",
+                  detailsURL: planFileUrl
+              }
+            }
+
+            // Only ask for manual approval when the build was manually launched by a human
+            // Note that we keep waiting with the current node agent as we want to keep the context
+            if (isBuildCauseUser){
+              stage("⏳ Waiting for User Manual Approval") {
+                input message: "Should we apply these changes to production?", ok: "🚢 Yes, ship it!"
+              }
+            }
+            if (!isBuildCauseTimer && isBuildOnProductionBranch) {
+              stage('🚢 Shipping Changes') {
+                sh makeCliCmd + ' deploy'
+              }
+            }
+          }
+        })
+      }
+    }
+
+    // Execute parallel stages from the map
+    parallel parallelStages
+  }
+}
+
+//   options {
+//     buildDiscarder(logRotator(numToKeepStr: '5'))
+//     timeout(time: 1, unit: 'HOURS')
+//     disableConcurrentBuilds()
+//     ansiColor('xterm')
+//   }
+
+def agentTemplate(containerImage, body) {
+  podTemplate(
+    // Custom YAML definition to enforce no svc account token (if Terraform uses kubernets, that would grant it a wrong access)
```

Comment:
```suggestion
    // Custom YAML definition to enforce no service account token (if Terraform uses kubernetes, it would grant it a wrong access)
```